### PR TITLE
Modify registry specs to adhere to spec naming conventions

### PR DESCRIPTION
**Pull request by mcquin**
Two files are testing `Chef::Win32::Registry`. Their spec paths should be `spec/*/win32/registry_spec.rb`.

### CODE REVIEW

File: spec/functional/win32/registry_spec.rb [hunk 1]
```diff
@@ -20,25 +20,6 @@
 require 'spec_helper'
 require 'chef/win32/registry'
 
-describe Chef::Resource::RegistryKey, :unix_only do
```
Comment (mcquin) [inline spec/functional/win32/registry_spec.rb]: This test is a duplicate of [this one](https://github.com/chef/chef/blob/master/spec/functional/resource/registry_spec.rb#L24-L41).

**Comment by mcquin** [comment]
This can be reviewed @chef/client-windows

**Comment by mwrock** [comment]
:+1:

**Comment by chefsalim** [comment]
Looks good :+1:

**Pull request merged**